### PR TITLE
Handle form urlencoded data #72

### PR DESCRIPTION
**Pull request by ryshep111**
Only use multer middleware on routes with multipart/form-data. Ensure that body parser is configured for url encoded form data. #72

### CODE REVIEW

File: handlers/_swagger.js [hunk 1]
```diff
@@ -171,6 +186,51 @@ function containsMultipartFormData(api) {
...
 }
+/**
+ * The current use of this function is to figure out whether body parser needs to be configured
+ * This is the case when there is an operation that consumes form data without a file
+ * If there is a file, then multer will be used to parse form data
+ * @param api
+ * @returns {boolean}
+ */
+function containsUrlEncodedFormData(api) {
+    if (_.indexOf(api.consumes, 'application/x-www-form-urlencoded') > -1) {
```
Comment (seanpk) [inline handlers/_swagger.js]: What would be the effect on our algorithm if there is a top-level `consumes` of `application/x-www-form-urlencoded`, and a particular method that overrides it to unset it?

We may need to carry the global value, if it exists, through the loop that is checking all paths.

**from the OpenAPI spec**

global `consumes`:
> A list of MIME types the APIs can consume. This is global to all APIs but can be overridden on specific API calls. Value MUST be as described under Mime Types.

path-specific `consumes`:
> A list of MIME types the operation can consume. This overrides the consumes definition at the Swagger Object. An empty value MAY be used to clear the global definition. Value MUST be as described under Mime Types.

Comment (seanpk) [inline handlers/_swagger.js]: ignore this concern if all that is required is to know whether any path can consume this MIME type

**Comment by seanpk** [comment]
I think we need some tests to validate that everything is working for both `application/x-www-form-urlencoded` and `multipart/form-data`

File: handlers/_swagger.js (hunk 1 above)
Comment (ryshep111) [inline handlers/_swagger.js]: yes, the purpose of that function is to figure out if any path consumes urlencoded data

File: handlers/_swagger.js [hunk 2]
```diff
@@ -68,6 +68,21 @@ exports.init = function (app, auth, config, logger, serviceLoader, swagger, call
             }
         }
 
+        if (containsUrlEncodedFormData(api)) {
+            var middlewareConfig = config.get('express').middleware;
+            var foundBodyParserInMiddlewareConfig = !(_.indexOf(middlewareConfig, 'body-parser') === -1 &&
```
Comment (ryshep111) [inline handlers/_swagger.js]: I just changed this to be < 0

File: handlers/_swagger.js [hunk 3]
```diff
@@ -70,14 +70,14 @@ exports.init = function (app, auth, config, logger, serviceLoader, swagger, call
 
         if (containsUrlEncodedFormData(api)) {
             var middlewareConfig = config.get('express').middleware;
-            var foundBodyParserInMiddlewareConfig = !(_.indexOf(middlewareConfig, 'body-parser') === -1 &&
-                _.indexOf(middlewareConfig, 'bodyParser') === -1);
+            var foundBodyParserInMiddlewareConfig = !(_.indexOf(middlewareConfig, 'body-parser') < 0 &&
```
Comment (seanpk) [inline handlers/_swagger.js]: covert to an OR without the external `!`

Comment (ryshep111) [inline handlers/_swagger.js]: done

**Comment by ryshep111** [comment]
test written

**Pull request merged**